sky/provision/kubernetes/utils.py

```diff
@@ -1006,12 +1008,14 @@ def get_ssh_proxy_command(
             private_key_path, ssh_jump_ip, ssh_jump_port=ssh_jump_port)
     else:
         ssh_jump_proxy_command_path = create_proxy_command_script()
+        current_context = get_current_kube_config_context_name()
```

Comment:
Should we add a `@functools.lru_cache()` to the `get_current_kube_config_context_name` to avoid the context name being changed when called multiple times?

Reply:
Hmm applying `lru_cache()` may cause issues when the user is running SkyPilot as a long-running wrapper in the background and switch kubecontext to change the current cluster being used.

E.g.,
```
import sky
sky.launch(...)
# user switches kubecontext externally
sky.launch(...) # Will use the old context for ssh which will fail 
```

Reply:
Is this an actual use case?  It seems to me we may want to keep the context / identity of a cloud always the same during the execution, otherwise, e.g. if people change the kubecontext during a `sky launch` is being executed in another terminal it may trigger some undefined behavior:
1. `sky launch`
2. During the pod provisioning before the SSH config is written, switch kubecontext in another terminal

After this, the cluster should be launched, but the ssh config may not be the correct kubecontext used for provisioning the pod?

This is a corner case, but we may want to fix it in the future? @romilbhardwaj